dev/releases/HelpLinks-to-JSON.g

```diff
@@ -0,0 +1,53 @@
+##  
+##  A utility for dumping links to all available manual sections 
+##  (for use with named manual references on web pages or in search
+##  utilities).
+##
+##  This is meant to be used with a plain vanilla GAP release, with
+##  only the bundled packages and no special user config.
+##
+
+LoadPackage("json");
+
+InstallMethod(_GapToJsonStreamInternal,
+[IsOutputStream, IsObject],
+function(o, x)
+    PrintTo(o, "null");
+end);
+
+out := OutputTextUser();
+r := rec();
+# load all books (without prompting the user for input)
+Perform(HELP_KNOWN_BOOKS[1], HELP_BOOK_INFO);;
+for x in Set(NamesOfComponents(HELP_BOOKS_INFO)) do # Any point doing Set?
```

Comment:
It would be nice to keep diffs between the `help-links.json` files of subsequent GAP versions as small as possible (although this is not essential). Therefore it would make sense for the books to be sorted in the output.

The question that I'm asking myself in this line is basically: does sorting the book names _here_ actually help to produce a sorted _output_?

The question arises because I am constructing my information as a record of records, and I have no idea whether there is any relationship between the _order_ in which I create record components, and the _order_ in which `GapToJsonStream` outputs record components to a stream.

Perhaps `GapToJsonStream` outputs record components: 
* in the order they were created (in which case the `Set` would help here, and we should also sort the entries of each book)
* randomly (in which case we can't do anything to help)
* in alphabetical order (in which case we don't need to do anything to get our desired behaviour; the `Set` should go).

I haven't yet experimented to see what it does.

Perhaps it would also be possible and sensible to get Python to sort everything when it reformats the JSON.

Reply:
After looking at this further, I am going to remove the `Set` since it doesn't affect the output. I have made a PR to the json package to help find a better solution https://github.com/gap-packages/json/pull/24